### PR TITLE
feat(dashboard) admin 3.0 create discounts form

### PR DESCRIPTION
**Pull request by fPolic**
**WHAT**
- added create discounts form

**NOTE**
- conditions sections will be added in a separate PR

### CODE REVIEW

**Comment by olivermrbl** [comment]
`Redemptions limit` and `Duration of the discount` are not included in the request to the backend

**Comment by fPolic** [comment]
> `Redemptions limit` and `Duration of the discount` are not included in the request to the backend

@olivermrbl - mixup with filed names, should be fixed now

File: packages/admin-next/dashboard/src/routes/discounts/create/components/create-discount-form/create-discount-form.tsx [hunk 1]
```diff
@@ -135,6 +135,8 @@ export const CreateDiscountForm = () => {
       return getDbAmount(amount, region!.currency_code)
     }
 
+    console.log(values)
```
Comment (olivermrbl) [inline packages/admin-next/dashboard/src/routes/discounts/create/components/create-discount-form/create-discount-form.tsx]: todo: rm

**Review by olivermrbl — APPROVED** [review]
Very nice! Will let @kasperkristensen glance over the implementation too

File: packages/admin-next/dashboard/src/routes/discounts/create/components/create-discount-form/create-discount-details.tsx [hunk 2]
```diff
@@ -11,56 +27,642 @@ type CreateDiscountPropsProps = {
...
+                        />
+                      </Form.Control>
+                    </div>
+                    <Form.Hint className="!mt-1">
+                      {t("discounts.startDateHint")}
+                    </Form.Hint>
+                    <Form.ErrorMessage />
+                  </Form.Item>
+                )}
+              />
```
Comment (kasperkristensen) [inline packages/admin-next/dashboard/src/routes/discounts/create/components/create-discount-form/create-discount-details.tsx:326]: This applies to all "sections" with a Switch. 

The intended design is that you click the switch to show the input. If you then disable the Switch it should hide the input and then the value should be undefined when submitting.

Then in the Edit form if the value exists on the Discount then it should be checked per default.

**Review by kasperkristensen — COMMENTED** [review]
LGTM, just the "Switch" sections need to be modified to have the same behaviour as everywhere else

File: packages/admin-next/dashboard/src/routes/discounts/create/components/create-discount-form/create-discount-details.tsx (hunk 2 above)
Comment (fPolic) [reply packages/admin-next/dashboard/src/routes/discounts/create/components/create-discount-form/create-discount-details.tsx:326]: Should be good now

**Review by olivermrbl — COMMENTED** [review]
nitpicking: If I flip the "Availability duration" switch but don't enter any details it will send them as `P0Y0M0DT0H0M0S` i.e. be expired.

**Comment by olivermrbl** [comment]
Nice, should be good now

**Pull request merged**